code/model/ErrorPage.php

```diff
@@ -113,7 +113,12 @@ public function requireDefaultRecords() {
 					}
 
 					// Ensure a static error page is created from latest error page content
+					$origReadingMode = static::get_reading_mode();
+					Versioned::reading_stage(Versioned::LIVE);
```

Comment:
This placement won't work, it needs to go before `$page = DataObject::get_one(...)` because the `udapteLink` method looks at the query params on the object which are set at query time...

Reply:
Oh yes you're right.

Reply:
Fixed and added tests.